src/pages/detail-page/sections/pc/MainSection.jsx

```diff
@@ -0,0 +1,149 @@
+import { useRef } from 'react';
+import { motion, useScroll, useTransform } from 'motion/react';
+import useElementHeight from '@hooks/useElementHeight';
+import DonationInfo from '@components/DonationInfo';
+import MainTitle from '../../components/MainTitle';
+import Button from '@components/common/Button';
+import DetailTitle from '@pages/detail-page/components/DetailTitle';
+import DetailContent from '@pages/detail-page/components/DetailContent';
+
+const MainSection = () => {
+  // ref를 통해 제목이 브라우저 가장 바닥에 위치한다. (absolute를 사용하면 제목 아래 본문과 이어지지 않는다.)
+  const [titleRef, top] = useElementHeight();
+  const scrollAreaRef = useRef(null);
+
+  // detail title 애니메이션
+  const detailTitleRef = useRef(null);
+  const { scrollYProgress: detailTitleScroll } = useScroll({
+    container: scrollAreaRef,
+    target: detailTitleRef,
+    offset: ['start end', 'start start'],
+  });
+  const detailTitleY = useTransform(
+    detailTitleScroll,
+    [0, 1],
+    ['-1000%', '0%'],
+  );
+  const detailTitleOpacity = useTransform(detailTitleScroll, [0, 0.2], [0, 1]);
+
+  // donation info 애니메이션
+  const donationInfoRef = useRef(null);
+  const { scrollYProgress: donationInfoScroll } = useScroll({
+    container: scrollAreaRef,
+    target: donationInfoRef,
+    offset: ['start end', 'start start'],
+  });
+  const donationInfoY = useTransform(
+    donationInfoScroll,
+    [0, 1],
+    ['-150%', '0%'],
+  );
+  const donationInfoOpacity = useTransform(
+    donationInfoScroll,
+    [0, 0.2],
+    [0, 1],
+  );
+
+  // background 애니메이션
+  const { scrollYProgress: backgroundScroll } = useScroll({
+    container: scrollAreaRef,
+    target: donationInfoRef,
+    offset: ['start end', 'start start'],
+  });
+  const backgroundY = useTransform(backgroundScroll, [0, 1], ['-300%', '0%']);
+  const backgroundOpacity = useTransform(backgroundScroll, [0, 0.2], [0, 1]);
```

Comment:
유사한 애니메이션 로직이 반복되고 있습니다.
커스텀 훅으로 추출하여 재사용성을 높일 수 있습니다.

```js
// useScrollAnimation.js

const useScrollAnimation = (containerRef, targetRef, yRange, opacityRange = [0, 0.2]) => {
  const { scrollYProgress } = useScroll({
    container: containerRef,
    target: targetRef,
    offset: ['start end', 'start start'],
  });
  
  const y = useTransform(scrollYProgress, [0, 1], yRange);
  const opacity = useTransform(scrollYProgress, [0, opacityRange[1]], [0, 1]);
  
  return { y, opacity };
};
```

다음과 같이 컴포넌트 내부에서 사용할 수 있습니다.
```suggestion
const detailTitleRef = useRef(null);
const donationInfoRef = useRef(null);

const detailTitleAnimation = useScrollAnimation(scrollAreaRef, detailTitleRef, ['-1000%', '0%']);
const donationInfoAnimation = useScrollAnimation(scrollAreaRef, donationInfoRef, ['-150%', '0%']);
const backgroundAnimation = useScrollAnimation(scrollAreaRef, donationInfoRef, ['-300%', '0%']);
```

Reply:
ref가 있기 때문에 어떻게 코드를 재사용해야 하나 고민하고 있었는데 도움주셔서 감사합니다!
훅으로 분리하면서 추가로 스크롤 감지 영역을 `undefined`로 초기값을 설정하면서, 전달받은 값이 없을때는 브라우저 화면 전체의 스크롤 감지하도록 해두어 다른 곳에서도 재사용할 수 있도록 해두었습니다!